### PR TITLE
Switch to GitHub Actions for CI

### PR DESCRIPTION
**Pull request by gazpachoking**
### Motivation for changes:
Try out github actions as a replacement for circleci
Hopefully we don't need to maintain custom docker images this way

#### To Do:

- [ ] Get it working
- [ ] Verify we like it better than circleci
- [ ] Add a release workflow

### CODE REVIEW

File: flexget/tests/conftest.py [hunk 1]
```diff
@@ -237,7 +238,7 @@ def filecopy(request):
                 else:
                     f.remove()
             except OSError as e:
-                print("couldn't remove %s: %s" % (f, e))
+                print ("couldn't remove %s: %s" % (f, e))
```
Comment (liiight) [inline flexget/tests/conftest.py]: added space by mistake?

Comment (gazpachoking) [reply flexget/tests/conftest.py]: Hmm, it happened twice in that commit, I remember seeing it in the diff after I ran black, but assumed it was fixing the extra space, not adding one...

Comment (gazpachoking) [reply flexget/tests/conftest.py]: Ahh, I must need to upgrade black on that machine. https://github.com/psf/black/issues/752

**Comment by gazpachoking** [comment]
Some stuff unrelated to actions was caught up in here. Merged that into develop directly and moved actions to new PR #2488

**Pull request closed without merging**